src/2d/prepc.f

```diff
@@ -26,16 +26,33 @@ subroutine prepc(level,nvar)
 c
 c ::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
 c
+      ! initialize to 0 in case no fine grids
+      listspStart(level) = 0
       maxsp  = 0
+      listsp(level) = maxsp
       mkid   = lstart(level+1)
  10   if (mkid .eq. 0) go to 20
          ikeep  = (node(ndihi,mkid)-node(ndilo,mkid)+1)/intratx(level)
          jkeep  = (node(ndjhi,mkid)-node(ndjlo,mkid)+1)/intraty(level)
          maxsp  = maxsp + 2*(ikeep+jkeep)
       mkid = node(levelptr,mkid)
       go to 10
- 20   listsp(level) = maxsp
-      if (maxsp .eq. 0) go to 99
+ 20   continue
+ 
+      !!! maxsp is enough storage for every coarse grid to have an adjustment by
+      !!! a corresponding fine grid flux sum. But not all will.
+      if (maxsp .eq. 0) go to 99  ! no fine grids, so no space needed. CHECK HOW THIS IS MARKED FOR COARSE GRIDS
```

Comment:
@mjberger: there's a comment in prepc.f:
CHECK HOW THIS IS MARKED FOR COARSE GRIDS
Is this something that still needs to be checked, or delete the comment?